thinc/backends/__init__.py

```diff
@@ -149,22 +155,14 @@ def contextvars_eq_thread_ops() -> bool:
     return False
 
 
-def _get_thread_state():
+def _get_thread_state() -> threading.local:
     """Get a thread-specific state variable that inherits from a global
     state when it's created."""
-    thread: threading.Thread = threading.current_thread()
-    if not hasattr(thread, "__local"):
-        thread.__local = _create_thread_local(_GLOBAL_STATE)
-    return thread.__local
-
-
-def _create_thread_local(
-    attrs: Dict[str, Any], local_class: Type[threading.local] = threading.local
-):
-    obj = local_class()
-    for name, value in attrs.items():
-        setattr(obj, name, value)
-    return obj
+    if not hasattr(_LOCAL_STATE, "initialized") or not _LOCAL_STATE.initialized:
```

Comment:
The other change that needs more careful review is here. This code didn't type check anymore, and it's hard to get to type check without ignores because we can't nilly-willy assign instance variables to classes. Also, it is not really necessary, since (paradoxically) we can use a global variable for thread-local storage, the storage will differ per thread.

Reply:
Looks good to me, though I'm not that across these details.